framework/src/service/ServiceReferenceBasePrivate.cpp

```diff
@@ -115,52 +126,42 @@ InterfaceMapConstPtr ServiceReferenceBasePrivate::GetServiceInterfaceMap(BundleP
 {
   InterfaceMapConstPtr s;
   if (!registration->available) return s;
-
   std::shared_ptr<ServiceFactory> serviceFactory;
-  int count = 0;
 
+  auto l = registration->Lock(); US_UNUSED(l);
+  if (!registration->available) return s;
+  serviceFactory = std::static_pointer_cast<ServiceFactory>(
+        registration->GetService_unlocked("org.cppmicroservices.factory"));
+
+  if(registration->dependents.end() == registration->dependents.find(bundle))
   {
-    auto l = registration->Lock(); US_UNUSED(l);
-    if (!registration->available) return s;
-    serviceFactory = std::static_pointer_cast<ServiceFactory>(
-          registration->GetService_unlocked("org.cppmicroservices.factory"));
-    count = registration->dependents[bundle];
+    registration->dependents.insert(std::make_pair(bundle, 0));
   }
 
   if (!serviceFactory)
   {
-    auto l = registration->Lock(); US_UNUSED(l);
-          s = registration->service;
-    if (s && !s->empty()) ++registration->dependents[bundle];
+    s = registration->service;
+    if (s && !s->empty())
+    {
+      ++registration->dependents.at(bundle);
+    }
+    return s;
+  }
+
+  if (0 == registration->dependents.at(bundle))
```

Comment:
Thanks for the suggestion, I've used it to simplify this logic. I will have an update to this PR shortly.

I prefer using ```at()``` over ```operator[]``` since I feel it makes the code easier to maintain. While all uses of ```at()``` are guaranteed not to throw, an exception being thrown would indicate a bug in the logic. If ```operator[]``` is used a potential bug of this kind would be harder to see.